src/main/java/appeng/tile/misc/TileVibrationChamber.java

```diff
@@ -244,22 +244,31 @@ private void eatFuel()
 			{
 				this.setBurnTime( this.getBurnTime() + newBurnTime );
 				this.setMaxBurnTime( this.getBurnTime() );
-				is.grow( -1 );
-				if( is.getCount() <= 0 )
+				if ( is.getItem() == Items.LAVA_BUCKET )
 				{
-					ItemStack container = ItemStack.EMPTY;
-
-					if( is.getItem() != Items.AIR && is.getItem().hasContainerItem( is ) )
-					{
-						container = is.getItem().getContainerItem( is );
-					}
-
-					this.inv.setStackInSlot( 0, container );
+					ItemStack bucket = new ItemStack( Items.LAVA_BUCKET );
+					this.inv.setStackInSlot( 0, bucket );
 				}
 				else
 				{
-					this.inv.setStackInSlot( 0, is );
+					is.grow( -1 );
+					if( is.getCount() <= 0 )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)